src/javascript/crypto/e2e/openpgp/asciiarmor_test.html

```diff
@@ -172,9 +172,29 @@
     assertEquals('hello world new', parsed.getBody());
     assertEquals('SHA1', parsed.getSignature().hashAlgorithm);
     console.log(e2e.openpgp.asciiArmor.encodeClearSign(parsed));
+
+    // Signed by user: "test 4" 1024-bit RSA key, ID 092DA808
+    var clearSignTrailingSpace =
+      '-----BEGIN PGP SIGNED MESSAGE----- \r\n' +
+      'Hash: SHA1 \r\n' +
+      ' \r\n' +
+      'hello world new \r\n' +
+      '-----BEGIN PGP SIGNATURE----- \r\n' +
+      'Version: GnuPG v1.4.11 (GNU/Linux) \r\n' +
+      ' \r\n' +
+      'iJwEAQECAAYFAlIny1sACgkQBv+crAktqAgPqgP/bG1lJAyVBa99rJTLn9RAJsC0 \r\n' +
+      '2d7hJBdEiwmakEzbGfWG48RM7mq4X+W3j36D8qeCWpYkJk+zd4g25TJ7DIwoBfCv \r\n' +
+      'J8S0vVfQ6FYTNHLuAggXs7ooLfovq2fP6iuVjwNX57JDxllHmS8tPAbIgIs+VRag \r\n' +
+      'hOia0yJgN0DoFSylakY= \r\n' +
+      '=qRFP \r\n' +
+      '-----END PGP SIGNATURE----- ';
+    parsed = e2e.openpgp.asciiArmor.parseClearSign(clearSignTrailingSpace);
+    assertEquals('hello world new', parsed.getBody());
+    assertEquals('SHA1', parsed.getSignature().hashAlgorithm);
+
     var invalid =
       '-----BEGIN PGP SIGNED MESSAGE-----\n' +
-      'Hash: SHA1\n \ntampered message.\n' +
+      // 'Hash: SHA1\n' + // missing hash algorithm
```

Comment:
@koto, the original code is actually NOT injecting an header, but adding an extra line of message. 

`invalid` is defined like so:
```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA1
 
tampered message.

hello world new
.
.
.
```

The ASCII armor format was rejected solely because of the single extra space in line 3. 

If we now tolerate the trailing space after every line, then the original ASCII armor will pass the new formatting/regex test. And in fact, it's the crypto logic to decide whether a "tampered message" sneaked into the message body. Parsing algorithm is there just to pick up the message. So, I updated the invalid armor testcase to test the regex of enforcing the use of Hash header.